packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -0,0 +1,386 @@
+
+    if (argResults['xcframework']) {
+      // Copy universal framework to variant directory.
+      final Directory armFlutterFrameworkDirectory = iPhoneModeDirectory.childDirectory(flutterFrameworkFileName);
+      final File armFlutterFrameworkBinary = armFlutterFrameworkDirectory.childFile('Flutter');
+      final File fatFlutterFrameworkBinary = fatFlutterFrameworkCopy.childFile('Flutter');
+      copyDirectorySync(fatFlutterFrameworkCopy, armFlutterFrameworkDirectory);
+
+      // Create iOS framework.
+      List<String> lipoCommand = <String>['xcrun', 'lipo', fatFlutterFrameworkBinary.path, '-remove', 'x86_64', '-output', armFlutterFrameworkBinary.path];
+
+      await processUtils.run(
+        lipoCommand,
+        workingDirectory: outputDirectory.path,
+        allowReentrantFlutter: false,
+      );
+
+      // Create simulator framework.
+      final Directory simulatorFlutterFrameworkDirectory = iPhoneSimulatorDirectory.childDirectory(flutterFrameworkFileName);
+      final File simulatorFlutterFrameworkBinary = simulatorFlutterFrameworkDirectory.childFile('Flutter');
+      copyDirectorySync(fatFlutterFrameworkCopy, simulatorFlutterFrameworkDirectory);
+
+      lipoCommand = <String>['xcrun', 'lipo', fatFlutterFrameworkBinary.path, '-thin', 'x86_64', '-output', simulatorFlutterFrameworkBinary.path];
+
+      await processUtils.run(
+        lipoCommand,
+        workingDirectory: outputDirectory.path,
+        allowReentrantFlutter: false,
+      );
+
+      // Create XCFramework from iOS and simulator frameworks.
+      final List<String> xcframeworkCommand = <String>[
+        'xcrun',
+        'xcodebuild',
+        '-create-xcframework',
+        '-framework', armFlutterFrameworkDirectory.path,
+        '-framework', simulatorFlutterFrameworkDirectory.path,
+        '-output', modeDirectory
+          .childFile('Flutter.xcframework')
+          .path
+      ];
+
+      await processUtils.run(
+        xcframeworkCommand,
+        workingDirectory: outputDirectory.path,
+        allowReentrantFlutter: false,
+      );
+    }
+
+    if (!argResults['universal']) {
+      fatFlutterFrameworkCopy.deleteSync(recursive: true);
+    }
+    status.stop();
+  }
+
+  Future<void> _produceAppFramework(BuildMode mode, Directory iPhoneModeDirectory, Directory modeDirectory) async {
+    const String appFrameworkName = 'App.framework';
+    final Directory destinationAppFrameworkDirectory = modeDirectory.childDirectory(appFrameworkName);
+
+    if (mode == BuildMode.debug) {
+      final Status status = logger.startProgress(' ├─Add placeholder App.framework for debug...', timeout: timeoutConfiguration.fastOperation);
+      await createStubAppFramework(destinationAppFrameworkDirectory);
+      status.stop();
+    } else {
+      await _produceAotAppFrameworkIfNeeded(mode, iPhoneModeDirectory, destinationAppFrameworkDirectory);
+    }
+
+    final FlutterProject flutterProject = FlutterProject.current();
```

Comment:
Creating a FlutterProject is actually kind of expensive, it would be worth doing it once at the beginning

Reply:
Ah ok. Injecting a single one around now.